### PR TITLE
FIX: Alter "Take Action" default behaviour to hide post

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit fixes an issue where clicking the default
"Take Action" option on a flag for a post doesn't always
end up with the post hidden.

This is because the "take_action" score bonus doesn’t take into account
the final score required to hide the post.

Especially with the `hide_post_sensitivity` site setting set to `low`
sensitivity, there is a likelihood the score needed to hide the post
won’t be reached.

Now, the default "Take Action" button has been changed to "Hide Post"
to reflect what is actually happening and the description has been
improved, and if "Take Action" is clicked we _always_ hide the post
regardless of score and sensitivity settings. This way the action reflects
expectations of the user.

image

### CODE REVIEW

**Review by nattsw — APPROVED** [review]
Approving as it fulfils what Ted says, but pending on James' response to merge.

**Comment by martin-brennan** [comment]
The post is still being unhidden straight away in some cases, I need to investigate this and further improve before merging.

File: spec/lib/post_action_creator_spec.rb [hunk 1]
```diff
@@ -271,18 +271,40 @@
   end
 
   describe "take_action" do
-    before { PostActionCreator.create(Fabricate(:user), post, :inappropriate) }
+    it "will hide the post" do
+      PostActionCreator
+        .new(Fabricate(:moderator), post, PostActionType.types[:spam], take_action: true)
+        .perform
+        .reviewable
+      expect(post.reload.hidden).to eq(true)
```
Comment (Drenmi) [inline spec/lib/post_action_creator_spec.rb:279]: Don't know if we have a convention for this, but the other tests here use the predicate matchers:

```suggestion
      expect(post.reload).to be_hidden
```

File: spec/lib/post_action_creator_spec.rb [hunk 2]
```diff
@@ -271,18 +271,40 @@
...
+
+      it "will agree with the old reviewable" do
+        reviewable =
+          PostActionCreator
+            .new(Fabricate(:moderator), post, PostActionType.types[:spam], take_action: true)
+            .perform
+            .reviewable
+        scores = reviewable.reviewable_scores
+        expect(scores[0]).to be_agreed
+        expect(scores[1]).to be_agreed
```
Comment (Drenmi) [inline spec/lib/post_action_creator_spec.rb:293]: I know this is not your code. Just pointing out the `#all` matcher which would probably read nicely here:

```suggestion
        expect(reviewable.reviewable_scores).to all(be_agreed)
```

**Review by Drenmi — APPROVED** [review]

**Pull request merged**